hack/olm-registry/olm-artifacts-template.yaml

```diff
@@ -132,7 +120,6 @@ objects:
           roleRef:
             kind: Role
             name: osd-metrics-exporter-watch-configmaps
-            namespace: openshift-config
```

Comment:
Why is the namespace removed from the `Role`?

I understand removing fields that don't show up in the schema, but a `Role` is namespaced, so I would expect this to stay in.

Reply:
Hey @bergmannf ,

The `roleRef` namespace will come from the current namespace

```
❯ oc explain rolebinding.roleRef
GROUP:      rbac.authorization.k8s.io
KIND:       RoleBinding
VERSION:    v1

FIELD: roleRef <RoleRef>

DESCRIPTION:
    RoleRef can reference a Role in the current namespace or a ClusterRole in
    the global namespace. If the RoleRef cannot be resolved, the Authorizer must
    return an error.
    RoleRef contains information that points to the role being used

FIELDS:
  apiGroup	<string> -required-
    APIGroup is the group for the resource being referenced

  kind	<string> -required-
    Kind is the type of resource being referenced

  name	<string> -required-
    Name is the name of resource being referenced
```

```
2023-12-18T08:04:00Z	INFO	controllers.ObjectSet	API status error with empty reason string	{"ObjectSet": "/managed-openshift-release-bundle-5f548cb8bb", "err": {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"failed to create typed patch object (openshift-config/osd-metrics-exporter-watch-configmaps; rbac.authorization.k8s.io/v1, Kind=RoleBinding): .roleRef.namespace: field not declared in schema","code":500}}
```

Reply:
It's just surprising that this is the only namespace being removed - because now it must be true that the namespace this is executed in is `openshift-config` (I am not sure it is - shouldn't it be `openshift-marketplace` as that is where the marketplace pods run?).

But if you're certain it's openshift-config ok - even though I'd prefer this MR to only remove the fields that are no longer supported, because I'm not sure this namespace removal won't break something.

Reply:
The namespace is being removed from the `roleRef` block, not the `metadata` block. Sorry this wasn't clear

```
           roleRef:
             kind: Role
             name: osd-metrics-exporter-watch-configmaps
-            namespace: openshift-config
             apiGroup: rbac.authorization.k8s.io
```

Reply:
```
❯ oc create -f roles.yaml --validate=true
role.rbac.authorization.k8s.io/osd-metrics-exporter-watch-configmaps created
Error from server (BadRequest): error when creating "roles.yaml": RoleBinding in version "v1" cannot be handled as a RoleBinding: strict decoding error: unknown field "roleRef.namespace"

❯ cat -p roles.yaml
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: osd-metrics-exporter-watch-configmaps
  namespace: openshift-config
rules:
  - apiGroups:
      - ""
    resources:
      - configmaps
    verbs:
      - get
      - list
      - watch
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: osd-metrics-exporter-watch-configmaps
  namespace: openshift-config
subjects:
  - kind: ServiceAccount
    name: osd-metrics-exporter
    namespace: openshift-osd-metrics
roleRef:
  kind: Role
  name: osd-metrics-exporter-watch-configmaps
  namespace: openshift-config
  apiGroup: rbac.authorization.k8s.io
```

if you remove the `namespace` from the `roleRef` at the bottom, then try again validation passes

Reply:
Ah I see - I accidentally misread the change, and thought it removed the `namespace` from the `metadata` field of the `Role`.
Sorry for the confusion - that was me just misreading what was actually changed - sorry.

Reply:
thank you for the review!!